crates/ruff_python_formatter/src/statement/suite.rs

```diff
@@ -124,7 +122,6 @@ impl FormatRule<Suite, PyFormatContext<'_>> for FormatSuite {
                 joiner.entry(statement, &statement.format());
             }
 
-            is_last_function_or_class_definition = is_current_function_or_class_definition;
```

Comment:
The idea was to only evaluate whether `statement` is a class or function definition once. Now we check the last and current in each iteration. But I guess the check is cheap enough